mvlearn/embed/dcca.py

```diff
@@ -0,0 +1,349 @@
+# Copyright 2019 NeuroData (http://neurodata.io)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# MIT License
+
+# Original work Copyright (c) 2016 Vahid Noroozi
+# Modified work Copyright 2019 Zhanghao Wu
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+
+from .base import BaseEmbed
+import torch
+from ..utils.utils import *
+
+
+class DCCA(BaseEmbed):
+    r"""
+    An implementation of Deep Canonical Correlation Analysis [#1DCCA]_ with
+    PyTorch. It computes projections into a common subspace in order to
+    maximize the correlation between pairwise projections into the subspace
+    from two views of data.
+
+    Parameters
+    ----------
+    input_size1 : int (positive)
+        The dimensionality of the input vectors in view 1.
+    input_size2 : int (positive)
+        The dimensionality of the input vectors in view 2.
+    outdim_size : int (positive), default=2
```

Comment:
is this n_components in gcca? I would go over gcca code and try to match up the variable names as much as possible since these are related algorithms.